File: 1328. Break a Palindrome.cpp
class Solution {
public:
    bool isPalin(const string& s) {
        int left = 0, right = s.length() - 1;
        while (left <= right) {
            if (s[left] != s[right]) return false;
            left++;
            right--;
        }
        return true;
    }

    string breakPalindrome(string palindrome) {
        int n = palindrome.length();
        if (n == 1) return "";

        bool foundNonA = false;
        string sb = "";

        for (int i = 0; i < n; i++) {
            if (palindrome[i] != 'a' && !foundNonA) {
                sb += 'a';
                foundNonA = true;
            } else {
                sb += palindrome[i];
            }
        }

        if (isPalin(sb) || !foundNonA) {
            sb = palindrome;
            sb[n - 1] = 'b';
        }

        return sb;
    }
};

File: 1328. Break a Palindrome.java
class Solution {
    public boolean isPalin(String s){
        int n = s.length();
        int left = 0, right = n - 1;
        while(left <= right){
            if(s.charAt(left) != s.charAt(right)) return false;
            left++;
            right--;
        }
        return true;
    }
    public String breakPalindrome(String palindrome) {
        boolean foundNonA = false;
        StringBuilder sb = new StringBuilder("");
        int n = palindrome.length();
        if(n == 1) return "";

        for(int i = 0; i < n; i++){
            if(palindrome.charAt(i) != 'a' && !foundNonA){
                foundNonA = true;
                sb.append('a');
            }
            else sb.append(palindrome.charAt(i));
        }
        // System.out.println(isPalin("aaccba"));
        if(isPalin(sb.toString()) || !foundNonA){
            sb = new StringBuilder(palindrome);
            sb.setCharAt(n - 1, 'b');
        }

        return sb.toString();
    }
}